src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -151,19 +153,19 @@ impl OpenIDConnectAuthProvider {
         })
     }
 
-    fn initialize_connection_if_needed(&self) -> KrillResult<()> {
-        let mut conn_guard = self.conn.write().unwrap(); // should never fail, better to panic and crash out if it does
+    async fn initialize_connection_if_needed(&self) -> KrillResult<()> {
+        let mut conn_guard = self.conn.write().await; // should never fail, better to panic and crash out if it does
```

Comment:
This comment seems to be outdated now as it no longer unwraps so I guess it can be removed.

Reply:
indeed, done